resource_cache/resource_cache_test.go

```diff
@@ -106,22 +110,14 @@ type Key string
 
 func TestObjectCache(t *testing.T) {
 
-	config := CacheConfig{
-		scheme:        scheme,
-		possibleGVKs:  make(map[schema.GroupVersionKind]bool),
-		protectedGVKs: make(map[schema.GroupVersionKind]bool),
-		logKey:        Key("bunk"),
-	}
-	var log logr.Logger
+	config := NewCacheConfig(scheme, nil, nil)
 
 	ctx := context.Background()
 	zapLog, _ := zap.NewDevelopment()
 
-	log = zapr.NewLogger(zapLog)
-
-	ctx = context.WithValue(ctx, Key("bunk"), &log)
+	log := zapr.NewLogger(zapLog)
 
-	oCache := NewObjectCache(ctx, k8sClient, &config)
+	oCache := NewObjectCache(ctx, k8sClient, log, config)
```

Comment:
Weird that we used to pass in `&config` even though `NewObjectCache` expects `*CacheConfig` and `NewCacheConfig` returns `*CacheConfig` - kinda confused why the previous version ever worked.

Reply:
Yeh - just rejigged a little